rpc/src/server/service.rs

```diff
@@ -516,6 +747,32 @@ mod fns {
         Ok(contract_address)
     }
 
+    #[inline]
+    fn address_to_contract_id(address: &str) -> Result<String, failure::Error> {
```

Comment:
This function is really unsafe because it's check data boundaries inside. Please refactor into a safe form.

Reply:
Can you elaborate please? I'm not sure how to refactor this into a safe form.

Reply:
For example, you have ```&address[2..4]``` on line 755, which will panic in case address length is less than 4. This function should check length of it's input data.

Reply:
Right, I get it now, Thanks